components/common-go/log/log.go

```diff
@@ -121,6 +129,30 @@ func (f *gcpFormatter) Format(entry *log.Entry) ([]byte, error) {
 		entry.Data["@type"] = "type.googleapis.com/google.devtools.clouderrorreporting.v1beta1.ReportedErrorEvent"
 	}
 
+	if !f.skipScrub {
+		for key, value := range entry.Data {
+			if key == "error" || key == "severity" || key == "message" || key == "time" || key == "serviceContext" || key == "context" {
+				continue
+			}
+			switch v := value.(type) {
+			case string:
+				entry.Data[key] = scrubber.Default.KeyValue(key, v)
+			case int, int8, int16, int32, int64, uint, uint8, uint16, uint32, uint64, float32, float64, complex64, complex128:
+				// no-op
+			case bool:
+				// no-op
+			case scrubber.Scrubbable:
+				err := v.Scrub(scrubber.Default)
+				if err != nil {
+					entry.Data[key] = "[redacted:failed]"
+					fmt.Fprintf(os.Stderr, "cannot scrub log entry: %v\n", err)
+				}
+			default:
+				entry.Data[key] = "[redacted:nested]"
+			}
+		}
+	}
```

Comment:
In go, the style is to have the least indentation on the longest part of the function. This means that we'd do

```
if f.skip {
	return f.JSONFormatter.Format(..)
}

for key, value := range entry.Data {
			if key == "error" || key == "severity" || key == "message" || key == "time" || key == "serviceContext" || key == "context" {
				continue
			}
			switch v := value.(type) {
			case string:
				entry.Data[key] = scrubber.Default.KeyValue(key, v)
			case int, int8, int16, int32, int64, uint, uint8, uint16, uint32, uint64, float32, float64, complex64, complex128:
				// no-op
			case bool:
				// no-op
			case scrubber.Scrubbable:
				err := v.Scrub(scrubber.Default)
				if err != nil {
					entry.Data[key] = "[redacted:failed]"
					fmt.Fprintf(os.Stderr, "cannot scrub log entry: %v\n", err)
				}
			default:
				entry.Data[key] = "[redacted:nested]"
			}
		}

return ...
```